### PR TITLE
Only show uncalled mutation that are in >=1 other sample of same patient

### PR DESCRIPTION
**Pull request by inodb**
Fix cbioportal/cbioportal#2613

Fixes showing of uncalled mutations on
http://triage.cbioportal.mskcc.org/case.do#/patient?caseId=P01-00001&navCaseIds=P01-00001&studyId=poetic

### CODE REVIEW

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 1]
```diff
@@ -720,8 +720,13 @@ export class PatientViewPageStore {
 
         for (const mutation of this.uncalledMutationData.result) {
             mutationId = PatientViewPageStore.getMutationId(mutation);
-            idToMutations[mutationId] = idToMutations[mutationId] || [];
-            idToMutations[mutationId].push(mutation);
+            let mutationList:Mutation[]|undefined = idToMutations[mutationId];
+            // only add uncalled mutation if it is called in another sample of
+            // this patient
+            if (mutationList !== undefined) {
+                idToMutations[mutationId] = idToMutations[mutationId];
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: i don't get this line. seems to be setting itself to itself.

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 2]
```diff
@@ -720,8 +720,13 @@ export class PatientViewPageStore {
 
         for (const mutation of this.uncalledMutationData.result) {
             mutationId = PatientViewPageStore.getMutationId(mutation);
-            idToMutations[mutationId] = idToMutations[mutationId] || [];
-            idToMutations[mutationId].push(mutation);
+            let mutationList:Mutation[]|undefined = idToMutations[mutationId];
+            // only add uncalled mutation if it is called in another sample of
+            // this patient
+            if (mutationList !== undefined) {
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: I think you could get rid of mutationList and just just write if (mutationId in idToMutations) i.e. use "in" operator

**Comment by alisman** [comment]
We should really have a test for this, since it's core logic.  It's difficult to test getters (@computed) in store.  We can either move function to helper or there is a way to access getter functions. Just need to research it.  Lets try to write a test for it.

**Comment by inodb** [comment]
Thanks @alisman , made the changes. Will try to make a test

**Comment by onursumer** [comment]
@inodb @alisman This is an outdated code, we need to do the same changes for `StoreUtils` in release branches, as well...

**Comment by inodb** [comment]
@onursumer @alisman ok, ill just make the test then for release

**Pull request merged**